src/legacy/core_plugins/kibana/public/management/index.js

```diff
@@ -70,6 +150,13 @@ uiModules
             item.active = `#${$location.path()}`.indexOf(item.url) > -1;
           });
         }
+
+        updateSidebar($scope.sections, $scope.section.id);
+        $scope.$on('$destroy', () => destroyReact(SIDENAV_ID));
+        management.addListener(() => updateSidebar(management.items.inOrder, $scope.section.id));
```

Comment:
I'm annoyed with this implementation and yet its a bit beyond the scope of this PR to create a full featured async section registration system. I've chosen to no longer pass `items` since it only passes the management section thats changed, not the complete set of management sections. Nor does the code triggering the listener have access to the global set of management sections. I suspect that a couple of clever lines of code could solve this but I've chosen to sidestep it.